.github/workflows/update-lock-files-pr.yml

```diff
@@ -0,0 +1,48 @@
+# Workflow to update lock files in a PR, triggered by specific PR comments
+name: Update lock files in PR
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  contents: write
```

Comment:
this means this action has write permissions on the repo, as a maintainer, which is pretty wild, so we should make sure we have the right safety mechanisms here. So the first thing to check is, if a user w/o write permissions submits a PR which modifies this file, or the corresponding python file doing the changes, is this action then using the PR's version or the version from `main`?

Reply:
I believe changes are always pushed to the PR branch instead of `main`, because we are getting the repository and ref to checkout using the following mechanism:

```bash
pr_info=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json headRefName,headRepository,headRepositoryOwner)
pr_head_ref=$(echo "$pr_info" | jq -r '.headRefName')
pr_head_repository=$(echo "$pr_info" | jq -r '.headRepositoryOwner.login + "/" + .headRepository.name')
```

And BTW actually all PRs have read and write permissions to this repo currently ([ref](https://github.com/scikit-learn/scikit-learn/settings/actions)); these lines are added because we may want to disable the global write permissions some day https://github.com/scikit-learn/scikit-learn/pull/29505#issuecomment-2248245812.

Reply:
What I mean is, if the user changes the code we have here, they can change it to targeting `main` and write to `main` or one of our branches on the main repo.